packages/webdriverio/src/utils/findStrategy.ts

```diff
@@ -229,3 +238,31 @@ export const findStrategy = function (selector: SelectorStrategy, isW3C?: boolea
 
     return { using, value }
 }
+
+const createRoleBaseXpathSelector = (role : any) => {
+    let locatorArr : Array<string> = []
+    roleElements.get(role)?.forEach((value : any)=>{
+        let locator : string
+        let tagname : string, tagAttribute : string | undefined, tagAttributevalue : string | number | undefined
+        tagname = value.name
+        if (value.attributes instanceof Array) {
+            value.attributes.forEach((val : any)=>{
```

Comment:
We should define a type for `val`

```suggestion
            value.attributes.forEach((val: any) => {
```

Reply:
so with this `tagAttribute` and `tagAttributevalue` are always whatever the last attributes of the element are?

Reply:
Hey @christian-bromann  Sorry could not get your question.

So As per the document of Aria-query 

elementRole will return Map of RoleKey which will contain set of HtmlElement Relationship & their Attribute Details

incase of Attribute => i think it should be only part of element and not specific order sequence 

https://github.com/A11yance/aria-query#role-to-element